core/variant/array.cpp

```diff
@@ -652,6 +652,26 @@ void Array::shuffle() {
 	}
 }
 
+void Array::swap(int p_first, int p_second) {
+	ERR_FAIL_COND_MSG(_p->read_only, "Array is in read-only state.");
+	const int s = _p->array.size();
+	if (s < 2) {
+		return;
```

Comment:
This would silently fail if the array has zero or one elements. I think that should cause an error.

I suppose an alternative approach would be to never cause an error, but return if it is succeded or not.

Reply:
Should we get an error, though? If an array has only one element, the method failing is actually the same as swapping that single element with itself; if the array is empty, we have nothing to swap at all. As an example, `shuffle` does the same, since the results are the same as well: no change in the array if `size < 2`.

Reply:
Oh you're right, trying to swap elements on size 0 or 1 arrays will likely result in out of range indices.